kcidb/test_db.py

```diff
@@ -19,194 +17,201 @@ def test_schemas_main():
-                     tests=["test:test:1"]),
-            parents=True,
-            children=True,
-            objects_per_report=10
-        )
-        return status
-    """)
+
     assert_executes(
-        json.dumps(empty), *argv,
-        driver_source=driver_source,
-        stdout_re=re.escape(
-            json.dumps(empty) + "\n" +
-            json.dumps(empty) + "\n"
-        )
-    )
+        json.dumps(empty), *argv)
 
 
-def test_load_main():
+def test_load_main(empty_database):
     """Check kcidb-db-load works"""
-    driver_source = textwrap.dedent("""
-        from unittest.mock import patch, Mock
-        from kcidb_io.schema import V4_1
-        client = Mock()
-        client.get_schema = Mock(return_value=((1, 0), V4_1))
-        with patch("kcidb.db.Client", return_value=client):
-            return function()
-    """)
-    argv = ["kcidb.db.load_main", "-d", "bigquery:project.dataset"]
 
-    assert_executes("", *argv, driver_source=driver_source)
-    assert_executes('{', *argv, driver_source=driver_source,
+    empty_database.is_initialized()
+    argv = ["kcidb.db.load_main", "-d", empty_database.database]
+
+    assert_executes("", *argv)
+    assert_executes('{', *argv,
                     status=1, stderr_re=".*JSONParseError.*")
-    assert_executes('{}', *argv, driver_source=driver_source,
+    assert_executes('{}', *argv,
                     status=1, stderr_re=".*ValidationError.*")
 
     empty = kcidb.io.SCHEMA.new()
+    assert_executes(json.dumps(empty), *argv)
 
-    driver_source = textwrap.dedent(f"""
-        from unittest.mock import patch, Mock
-        from kcidb_io.schema import V4_1
-        client = Mock()
-        client.get_schema = Mock(return_value=((1, 0), V4_1))
-        client.load = Mock()
-        with patch("kcidb.db.Client", return_value=client) as Client:
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        client.load.assert_called_once_with({repr(empty)})
-        return status
-    """)
-    assert_executes(json.dumps(empty), *argv,
-                    driver_source=driver_source)
-
-    driver_source = textwrap.dedent(f"""
-        from unittest.mock import patch, Mock, call
-        from kcidb_io.schema import V4_1
-        client = Mock()
-        client.get_schema = Mock(return_value=((1, 0), V4_1))
-        client.load = Mock()
-        with patch("kcidb.db.Client", return_value=client) as Client:
-            status = function()
-        Client.assert_called_once_with("bigquery:project.dataset")
-        assert client.load.call_count == 2
-        client.load.assert_has_calls([call({repr(empty)}),
-                                      call({repr(empty)})])
-        return status
-    """)
-    assert_executes(json.dumps(empty) + json.dumps(empty), *argv,
-                    driver_source=driver_source)
+    assert_executes(json.dumps(empty) + json.dumps(empty), *argv)
```

Comment:
Would be good to also load some actual data (e.g. the same data used by other tests), and verify it gets loaded, but we can get it one step at a time, and do that in the next PR.